ee/clickhouse/queries/column_optimizer.py

```diff
@@ -19,9 +19,10 @@ class ColumnOptimizer:
     This speeds up queries since clickhouse ends up selecting less data.
     """
 
-    def __init__(self, filter: Union[Filter, PathFilter, RetentionFilter], team_id: int):
+    def __init__(self, filter: Union[Filter, PathFilter, RetentionFilter], team_id: int, is_actor_query: bool = False):
```

Comment:
Q: Why is this needed?

In general we want to avoid coupling column_optimizer with other modules even tighter. Since everything already depends on it, we end up with a two-way dependency which is just hell to debug.

Reply:
Didn't like the coupling either but seemed like the most streamlined way to add in the flag that would tell the querybuilder to add the group join in

Reply:
nvm. was looking at this the wrong way. after we decided to use postgres data, very little changes were needed